### PR TITLE
fix: remove token check for report

### PR DESCRIPTION
**Pull request by thomasrockhu-codecov**

### CODE REVIEW

**Review by joseph-sentry — APPROVED** [review]

File: codecov_cli/services/report/__init__.py [hunk 1]
```diff
@@ -80,7 +80,13 @@ def create_report_results_logic(
 def send_reports_result_request(
     commit_sha, report_code, encoded_slug, service, token, enterprise_url
 ):
-    headers = get_token_header_or_fail(token)
+    tokenless = os.environ.get("TOKENLESS")
```
Comment (joseph-sentry) [inline codecov_cli/services/report/__init__.py:83]: I don't think the endpoint here supports this

File: codecov_cli/services/report/__init__.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from codecov_cli.helpers.config import CODECOV_API_URL
 from codecov_cli.helpers.encoder import decode_slug, encode_slug
 from codecov_cli.helpers.request import (
-    get_token_header_or_fail,
+    get_token_header,
```
Comment (joseph-sentry) [inline codecov_cli/services/report/__init__.py:11]: we should still keep get_token_header_or_fail here

**Review by joseph-sentry — CHANGES_REQUESTED** [review]

**Review by joseph-sentry — APPROVED** [review]

File: codecov_cli/services/report/__init__.py (hunk 2 above)
Comment (giovanni-guidini) [reply codecov_cli/services/report/__init__.py:11]: Your comment is conflicting with the code

**Review by giovanni-guidini — APPROVED** [review]

File: codecov_cli/services/report/__init__.py (hunk 2 above)
Comment (joseph-sentry) [reply codecov_cli/services/report/__init__.py:11]: there was an old version of the PR where `get_token_header_or_fail` was being deleted, i meant that we should keep both

**Pull request merged**